### PR TITLE
Snmp access

### PR DESCRIPTION
**Pull request by SourceDoctor**
Access Rules for snmp Requests (equal to http_access)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -175,6 +175,36 @@ Adds a squid.conf line
 http_access allow our_networks hosts
 ```
 
+### Defined Type squid::snmp\_access
+Defines [snmp_access entries](snmp://www.squid-cache.org/Doc/config/snmp_access/) for a squid server.
```
Comment (alexjfisher) [inline README.md]: That's a funny looking URL :)

File: README.md [hunk 2]
```diff
@@ -175,6 +175,36 @@ Adds a squid.conf line
...
+```puppet
+squid::snmp_access { 'our_monitoring hosts':
+  action => 'allow',
+}
+```
+
+Adds a squid.conf line 
+
+```
+# snmp_access fragment for out_monitoring hosts
```
Comment (alexjfisher) [inline README.md]: `out_monitoring hosts` typo?

File: manifests/init.pp [hunk 3]
```diff
@@ -17,6 +17,7 @@
   Optional[String]  $coredump_dir                  = $squid::params::coredump_dir,
   Optional[Hash]    $extra_config_sections         = {},
   Optional[Hash]    $http_access                   = $squid::params::http_access,
+  Optional[Hash]    $snmp_access                   = $squid::params::snmp_access,
```
Comment (alexjfisher) [inline manifests/init.pp]: Looks like you've forgotten to set a default in `params.pp`

Comment (alexjfisher) [reply manifests/init.pp]: and that's why the tests are failing.

**Comment by traylenator** [comment]
params.pp has now been created.

**Comment by traylenator** [comment]
@SourceDoctor thanks. This snmp one is one I had been intending to add for a while.

**Pull request merged**